service/init.go

```diff
@@ -304,6 +309,25 @@ func (pcf *PCF) Exec(c *cli.Context) error {
 	return err
 }
 
+func (pcf *PCF) StartKeepAliveTimer(nfProfile models.NfProfile) {
+	if nfProfile.HeartBeatTimer == 0 {
+		nfProfile.HeartBeatTimer = 30
+	}
+	//stop keepAliveTimer if its running
+	pcf.StopKeepAliveTimer()
+	logger.InitLog.Infof("Started KeepAlive Timer: %v sec", nfProfile.HeartBeatTimer)
+	//AfterFunc starts timer and waits for KeepAliveTimer to elapse and then calls pcf.UpdateNF function
+	KeepAliveTimer = time.AfterFunc(time.Duration(nfProfile.HeartBeatTimer)*time.Second, pcf.UpdateNF)
+}
+
+func (pcf *PCF) StopKeepAliveTimer() {
+	if KeepAliveTimer != nil {
+		logger.InitLog.Infof("Stopped KeepAlive Timer.")
+		KeepAliveTimer.Stop()
+		KeepAliveTimer = nil
```

Comment:
Just Check- I think "KeepAliveTimer" is accessed from multiple go-routines, u should protect read/write via mutex